### ISSUE TITLE
numpy 2.0 causes issues

### ISSUE THREAD

**Issue by MattyD797** [xaviermouy/FishSound_Finder#3]
The most up to date numpy version or >2.0 causes problems 

Version == 0.1.0
OS == Windows 10
python == 3.9

I was getting on the first lines the following when running fishsound_finder:

```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
```

For the full traceback:

````
>fishsound_finder

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ralph\miniconda3\envs\findfish4\Scripts\fishsound_finder.exe\__main__.py", line 4, in <module>
    from fishsound_finder.detector import main
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\fishsound_finder\detector.py", line 37, in <module>
    silence_tensorflow()
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\silence_tensorflow\silence_tensorflow.py", line 79, in silence_tensorflow
    import tensorflow as tf  # pylint: disable=import-outside-toplevel
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\__init__.py", line 37, in <module>
    from tensorflow.python.eager import context
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\eager\context.py", line 35, in <module>
    from tensorflow.python.client import pywrap_tf_session
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\client\pywrap_tf_session.py", line 19, in <module>
    from tensorflow.python.client._pywrap_tf_session import *
AttributeError: _ARRAY_API not found

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ralph\miniconda3\envs\findfish4\Scripts\fishsound_finder.exe\__main__.py", line 4, in <module>
    from fishsound_finder.detector import main
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\fishsound_finder\detector.py", line 37, in <module>
    silence_tensorflow()
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\silence_tensorflow\silence_tensorflow.py", line 79, in silence_tensorflow
    import tensorflow as tf  # pylint: disable=import-outside-toplevel
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\__init__.py", line 42, in <module>
    from tensorflow.python import data
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\__init__.py", line 21, in <module>
    from tensorflow.python.data import experimental
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\__init__.py", line 95, in <module>
    from tensorflow.python.data.experimental import service
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\service\__init__.py", line 387, in <module>
    from tensorflow.python.data.experimental.ops.data_service_ops import distribute
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\ops\data_service_ops.py", line 23, in <module>
    from tensorflow.python.data.experimental.ops import compression_ops
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\ops\compression_ops.py", line 16, in <module>
    from tensorflow.python.data.util import structure
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\util\structure.py", line 22, in <module>
    from tensorflow.python.data.util import nest
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\util\nest.py", line 36, in <module>
    from tensorflow.python.framework import sparse_tensor as _sparse_tensor
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\sparse_tensor.py", line 24, in <module>
    from tensorflow.python.framework import constant_op
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\constant_op.py", line 25, in <module>
    from tensorflow.python.eager import execute
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\eager\execute.py", line 23, in <module>
    from tensorflow.python.framework import dtypes
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\dtypes.py", line 26, in <module>
    from tensorflow.python.lib.core import _pywrap_bfloat16
AttributeError: _ARRAY_API not found
ImportError: numpy.core._multiarray_umath failed to import
ImportError: numpy.core.umath failed to import
Traceback (most recent call last):
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ralph\miniconda3\envs\findfish4\Scripts\fishsound_finder.exe\__main__.py", line 4, in <module>
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\fishsound_finder\detector.py", line 37, in <module>
    silence_tensorflow()
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\silence_tensorflow\silence_tensorflow.py", line 79, in silence_tensorflow
    import tensorflow as tf  # pylint: disable=import-outside-toplevel
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\__init__.py", line 42, in <module>
    from tensorflow.python import data
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\__init__.py", line 21, in <module>
    from tensorflow.python.data import experimental
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\__init__.py", line 95, in <module>
    from tensorflow.python.data.experimental import service
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\service\__init__.py", line 387, in <module>
    from tensorflow.python.data.experimental.ops.data_service_ops import distribute
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\ops\data_service_ops.py", line 23, in <module>
    from tensorflow.python.data.experimental.ops import compression_ops
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\experimental\ops\compression_ops.py", line 16, in <module>
    from tensorflow.python.data.util import structure
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\util\structure.py", line 22, in <module>
    from tensorflow.python.data.util import nest
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\data\util\nest.py", line 36, in <module>
    from tensorflow.python.framework import sparse_tensor as _sparse_tensor
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\sparse_tensor.py", line 24, in <module>
    from tensorflow.python.framework import constant_op
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\constant_op.py", line 25, in <module>
    from tensorflow.python.eager import execute
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\eager\execute.py", line 23, in <module>
    from tensorflow.python.framework import dtypes
  File "C:\Users\ralph\miniconda3\envs\findfish4\lib\site-packages\tensorflow\python\framework\dtypes.py", line 29, in <module>
    _np_bfloat16 = _pywrap_bfloat16.TF_bfloat16_type()
TypeError: Unable to convert function return value to a Python type! The signature was
        () -> handle

**Comment by MattyD797** [comment]
For now and others if they run into this problem, I was able to work around it by installing numpy 1.26.4 via the following in a conda environment:

```
pip install numpy==1.26.4
```

**Comment by xaviermouy** [comment]
@MattyD797 thanks for reporting this issue. The requirements.txt file has been edited to explicitly include all dependencies and associated versions. This will hopefully avoid this type of issues in the future.  I released these changes under v0.1.1.
